dirsrvtests/tests/suites/plugins/attruniq_test.py

```diff
@@ -323,5 +491,500 @@ def test_exclude_subtrees(topology_st):
+    :expectedresults:
+        1. Plugin configured and enabled successfully
+        2. ModRDN operation fails with CONSTRAINT_VIOLATION
+    """
+
+    log.info('Setup attribute uniqueness plugin for "cn" attribute')
+    attruniq.add_unique_subtree(ACTIVE_DN)
+    attruniq.enable()
+    topology_st.standalone.restart()
+
+    log.info('Verify that uniqueness rule is enforced')
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        active_user_2.rename(f'cn={ACTIVE_USER_1_CN}', deloldrdn=False)
+
+
+def test_multiple_containers_add(topology_st, attruniq, containers,
+                                  active_user_1, stage_user_1):
+    """Test ADD operations with attribute uniqueness across multiple containers.
+
+    This test verifies that when multiple containers are configured for
+    attribute uniqueness, the plugin enforces uniqueness within each container
+    separately, but allows duplicate values across different containers.
+
+    :id: 400cdf54-499a-4c23-91f5-835f44835e3e
+    :setup: Standalone instance with containers and users in different containers
+    :steps:
+        1. Configure plugin for 'cn' attribute in both active and stage containers
+        2. Try to add user with duplicate 'cn' in active container (should fail)
+        3. Try to add user with duplicate 'cn' in stage container (should fail)
+        4. Try to add user in active container with 'cn' matching stage user (should succeed)
+        5. Try to add user in stage container with 'cn' matching active user (should succeed)
+    :expectedresults:
+        1. Plugin configured successfully
+        2. Duplicate in active container fails with CONSTRAINT_VIOLATION
+        3. Duplicate in stage container fails with CONSTRAINT_VIOLATION
+        4. Cross-container duplicate in active succeeds
+        5. Cross-container duplicate in stage succeeds
+    """
+
+    log.info('Setup attribute uniqueness plugin for "cn" attribute')
+    attruniq.add_unique_subtree(ACTIVE_DN)
+    attruniq.add_unique_subtree(STAGE_DN)
+    attruniq.enable()
+    topology_st.standalone.restart()
+
+    log.info("Verify both subtrees apply attruniqueness separately")
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        create_active_user(topology_st, uid=ACTIVE_USER_2_CN, cn=[ACTIVE_USER_1_CN, ACTIVE_USER_2_CN])
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        create_stage_user(topology_st, uid=STAGE_USER_2_CN, cn=[STAGE_USER_1_CN, STAGE_USER_2_CN])
+
+    log.info("Verify that the subtrees do not enforce uniqueness across each other")
+    try:
+        active_user_2 = create_active_user(topology_st, uid=ACTIVE_USER_2_CN,
+                                           cn=[STAGE_USER_1_CN, ACTIVE_USER_2_CN])
+        stage_user_2 = create_stage_user(topology_st, uid=STAGE_USER_2_CN,
+                                         cn=[ACTIVE_USER_1_CN, STAGE_USER_2_CN])
+    finally:
+        active_user_2.delete()
+        stage_user_2.delete()
```

Comment:
Possible UnboundLocalError.
If either create_* unexpectedly raises, the finally references will crash the test.